src/main/java/com/dev/cinema/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,36 @@
+package com.dev.cinema.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import com.dev.cinema.model.Role;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).get();
```

Comment:
let's use orElseThrow() instead of get and throw UserNotFoundException

Reply:
Changed.